README.md

```diff
@@ -1,16 +1,817 @@
+
+Let's look at this solution code from the previous lesson:
+
+```ruby
+def map_to_negativize(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] * -1 ) # <== Unique ork
+    i += 1
+  end
+  return new
+end
+
+def map_to_no_change(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] ) # <== Unique work
+    i += 1
+  end
+  return new
+end
+
+def map_to_double(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] * 2 ) # <== Unique work
+    i += 1
+  end
+  return new
+end
+
+def map_to_square(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] * source_array[i] ) # <== Unique work
+    i += 1
+  end
+  return new
+end
+```
+
+As you can see, the only thing that varies between these methods is the line
+which we've labeled with Unique work. When you learned to create methods, you
+probably learned that the way to _abstract_ the method is to pass in the stuff
+that varies as an _argument_ in the _call_ of the method. We'd like to do the
+same thing here.
+
+But in Ruby we know we can pass `String`s, `Array`s, `Hash`es, and `Float`s can
+we pass the abstraction of "work to do later?" With _blocks_ that's exactly
+what we do! We pass work, to be executed later, as an _implicit_ argument.
+
+## State Two Ways of Constructing Blocks
+
+We denote blocks in Ruby using two syntaxes:
+
+1. Curly brace `{}`
+2. `do...end`
+
+We use curly brace when the work inside is generally only one expression. Since
+expressions in Ruby _always_ return a value, the return value of the block will
+the value of that single expression.
+
+We use `do...end` when the work takes multiple lines. Just like methods,
+whatever is on the last line of a `do...end` block will be the return value of
+the block.
+
+### Code Writing Tip
+
+It's wise to start out with a `do...end` blocks because, if you need to add
+debugging data `p "this should be #{i}"`, you won't have any problems. Once
+you're sure you've gotten your implementation correct, you can use the shorter,
+and more elegant `{}` format.
+
+If you're stubborn and refuse this advise, if you need to include multiple
+expressions inside of a `{}`, you'll need to separate the expressions with `;`.
+On the other hand, it's hard to see how an expression like `-1 * x` could go
+wrong, so _maybe_ in some cases you can start with `{}`.
+
+## Pass a Block to a Method
+
+We pass a block to a method by including either a `{}` or a `do...end` block
+after our call to the method.
+
+```ruby
+method_using_block { puts "hi" }
+method_using_block do
+  puts "hi"
+end
+```
+
+## Execute a Code Passed as a Block Inside a Method
+
+When we pass arguments into a method, we have _parameters_ that allow us to
+refer to them:
+
+```ruby
+def razzle(title, name, byname)
+end
+
+razzle("King", "Hobo", "the Magnificent")
```

Comment:
I don't really mind this example domain specifically, but I do have a thought - is there a sort of cognitive burden in weird references? With an already difficult concept like `yield`, I think it may be easier for students to digest if we are delivering the idea in a mayonnaise and cheese sandwich on white bread. Better to be clear and understood, even if we are a little more bland and boring, if you get my meaning. Maybe expanding on a domain we've already covered or going with a dead simple, no extra vocab code example is better here

Reply:
ruby-enumerables-generalized-map-and-reduce-lab

changes it to sandwiches.